pkg/sdk/security_integrations_def.go

```diff
@@ -63,6 +97,97 @@ const (
 	OauthSecurityIntegrationClientTableauServer  OauthSecurityIntegrationClientOption = "TABLEAU_SERVER"
 )
 
+var AllOauthSecurityIntegrationClients = []OauthSecurityIntegrationClientOption{
+	OauthSecurityIntegrationClientLooker,
+	OauthSecurityIntegrationClientTableauDesktop,
+	OauthSecurityIntegrationClientTableauServer,
+}
+
+func ToOauthSecurityIntegrationClientOption(s string) (OauthSecurityIntegrationClientOption, error) {
+	s = strings.ToUpper(s)
+	switch s {
+	case "LOOKER":
+		return OauthSecurityIntegrationClientLooker, nil
+	case "TABLEAU_DESKTOP":
+		return OauthSecurityIntegrationClientTableauDesktop, nil
+	case "TABLEAU_SERVER":
+		return OauthSecurityIntegrationClientTableauServer, nil
+	default:
+		return "", fmt.Errorf("invalid OauthSecurityIntegrationClientOption: %s", s)
+	}
+}
+
+type Saml2SecurityIntegrationSaml2ProviderOption string
+
+const (
+	Saml2SecurityIntegrationSaml2ProviderOkta   Saml2SecurityIntegrationSaml2ProviderOption = "OKTA"
+	Saml2SecurityIntegrationSaml2ProviderAdfs   Saml2SecurityIntegrationSaml2ProviderOption = "ADFS"
+	Saml2SecurityIntegrationSaml2ProviderCustom Saml2SecurityIntegrationSaml2ProviderOption = "CUSTOM"
+)
+
+var AllSaml2SecurityIntegrationSaml2Providers = []Saml2SecurityIntegrationSaml2ProviderOption{
+	Saml2SecurityIntegrationSaml2ProviderOkta,
+	Saml2SecurityIntegrationSaml2ProviderAdfs,
+	Saml2SecurityIntegrationSaml2ProviderCustom,
+}
+
+func ToSaml2SecurityIntegrationSaml2ProviderOption(s string) (Saml2SecurityIntegrationSaml2ProviderOption, error) {
+	s = strings.ToUpper(s)
+	switch s {
+	case "OKTA":
+		return Saml2SecurityIntegrationSaml2ProviderOkta, nil
+	case "ADFS":
+		return Saml2SecurityIntegrationSaml2ProviderAdfs, nil
+	case "CUSTOM":
+		return Saml2SecurityIntegrationSaml2ProviderCustom, nil
+	default:
+		return "", fmt.Errorf("invalid Saml2SecurityIntegrationSaml2ProviderOption: %s", s)
+	}
+}
+
+type Saml2SecurityIntegrationSaml2RequestedNameidFormatOption string
+
+const (
+	Saml2SecurityIntegrationSaml2RequestedNameidFormatUnspecified                Saml2SecurityIntegrationSaml2RequestedNameidFormatOption = "urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified"
```

Comment:
wwow, do we really need those?

Reply:
Yes because you can't set this field to an arbitrary value - check [docs](https://docs.snowflake.com/en/sql-reference/sql/create-security-integration-saml2#optional-parameters).